extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -126,7 +126,7 @@ class MultipleBackendTask extends ReplicateObject {
         const params = {
             bucket: sourceEntry.getBucket(),
             objectKey: sourceEntry.getObjectKey(),
-            versionId: sourceEntry.getEncodedVersionId(),
+            versionId: sourceEntry.getEncodedVersionId() || 'null',
```

Comment:
should we instead modify `getEncodedVersionId()` to return 'null' ?
(i.e. return an actual version id...)

Reply:
....this is in arsenal :-(

Reply:
no cause the function can also be used for non versioned buckets, in that case it should return undefined

Reply:
ok, let's leave it for now.

but it seems it should be the other way around : getEncodedVersionId() should alwats return the version id (which may be "null") and it would be up to the caller to check if the bucket is versionned → call getEncodedVersionId() or non-versionned → completely ignore the version id.

What do you think? Should we create a TechDebt (cleanup) ticket ?

Reply:
For non-versioned buckets, putMetadata("null") should update the object master, so we should be good to return "null" if bucket is non-versioned.